### ISSUE TITLE
Does the the ring-of-rings branch of Horovod support the pytorch computing framework?

### ISSUE THREAD

**Issue by Frank00001** [horovod/horovod#2573]
I have observed that the "ring-of-rings" version of horovod only supports tensorflow. How can I adapt to the pytorch framework?

**Comment by chongxiaoc** [comment]
That branch is behind master by 700 commits, how about just cherry-picking `ring-of-rings` commit on top of `master`? Though I don't understand `ring-of-rings` features.

**Comment by Frank00001** [comment]
> That branch is behind master by 700 commits, how about just cherry-picking `ring-of-rings` commit on top of `master`? Though I don't understand `ring-of-rings` features.
The master branch only supports the ring-allreduce algorithm, while the ring-of-rings branch supports hierarchical allreduce algorithm.

**Comment by tgaddair** [comment]
Hey @Frank00001, feel free to update the branch to work with the most recent Horovod version, contributions always welcome!  One thing I would point out with this branch is we did not observe any performance improvement using this approach ourselves.  In practice, we've found that NCCL is better suited to make these kinds of topology optimizations.  But happy to chat about this is you have some ideas.
